### PR TITLE
OGM-571 Honor HBM2DDL for creation and validation

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-571

Tihs is built using the initial work made by @gunnarmorling.

After this PR the creation of the schema follows the HBM2DDL  properties instead of the `create_database` property. `create_database` can still be used to create the database with the datastores when it makes sense.

I'm not very satisfied with this PR, there are a couple of things that are missing:
1. The drop command because our SchemDefiner interface does not have a method to drop the schema
2. The validation is a bit confusing: SchemaDefiner validate both the values used to map the entities and the content of the datastore (It depends which datastore we consider). This part requires some refactoring.

But even so, I think at the moment this patch could be integrated and we can continue after a new pair of eyes reviewed it.

Cheers

### CODE REVIEW

**Comment by gsmet** [comment]
@DavideD the build fails because of a checkstyle failure:
/mnt/jenkins-workdir/workspace/hibernate-ogm-PR/core/src/main/java/org/hibernate/ogm/schema/impl/NoSqlSchemaManagementTool.java:31: error: White spaces at the end of line

**Comment by DavideD** [comment]
> @DavideD the build fails because of a checkstyle failure:

Damn last minute change, should work now. Sorry

**Comment by gsmet** [comment]
I know your pain :). Thanks!

File: infinispan-remote/src/test/java/org/hibernate/ogm/datastore/infinispanremote/test/initialize/CacheNotDefinedTest.java [hunk 1]
```diff
@@ -63,7 +65,7 @@ private void createFactory(Class<?>... entities) {
 		settings.put( OgmProperties.DATASTORE_PROVIDER, "infinispan_remote" );
 		settings.put( InfinispanRemoteProperties.CONFIGURATION_RESOURCE_NAME, "hotrod-client-testingconfiguration.properties" );
 		// This is the important option
-		settings.put( OgmProperties.CREATE_DATABASE, false );
+		settings.put( AvailableSettings.HBM2DDL_DATABASE_ACTION, Action.VALIDATE );
```
[the remark this answers is not in the document]
Comment (DavideD) [reply infinispan-remote/src/test/java/org/hibernate/ogm/datastore/infinispanremote/test/initialize/CacheNotDefinedTest.java]: As far as I understand they are the same thing, one is JPA standard and the other is hibernate specific.  No need to get confused :)

**Comment by DavideD** [comment]
I'. closing this one and create a new one with the changes we discussed and the drop for MongoDB

**Pull request closed without merging**